searchlib/src/vespa/searchlib/tensor/tensor_buffer_store.h

```diff
@@ -72,6 +72,13 @@ class TensorBufferStore : public TensorStore
         auto buf = _array_store.get(ref);
         return _ops.get_serialized_tensor_ref(buf);
     }
+    void prefetch_vectors(EntryRef ref) const noexcept {
+        const auto buf = _array_store.get(ref);
+
+        for (std::size_t i = 0; i < buf.size(); i += 64) {
+            __builtin_prefetch(buf.data() + i);
+        }
```

Comment:
I'm a bit concerned that this ends prefetching _everything_ across all subspaces, which can pollute the caches with stuff that will not be used (especially L1d)...

The first cache line of `buf` holds the subspace header, so maybe that should be prefetched first, then prefetch the actual subspace afterwards.